src/t8_cmesh/t8_cmesh_readmshfile.c

```diff
@@ -173,6 +173,63 @@ t8_msh_file_node_compare (const void *node_a, const void *node_b,
   return Node_a->index == Node_b->index;
 }
 
+/* Read an open msh-file and checks whether the MeshFormat-Version is supported by t8code or not. */
+static void
+t8_cmesh_check_version_of_msh_file (FILE * fp)
+{
+
+  char               *line = (char *) malloc (1024);
```

Comment:
There is no `free` to this `malloc`.
Have a look at the other functions and their `goto die...` statements.
You have several points in the functions where you must abort and return. For example when you check the return value of sscanf. In this case, you must free the allocated memory and return. Both currently does not happen.